flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/ReadOnlySlicedNetworkBuffer.java

```diff
@@ -75,12 +75,12 @@ private ReadOnlySlicedNetworkBuffer(ByteBuf buffer, int index, int length, int m
 
 	@Override
 	public ByteBuf unwrap() {
-		return super.unwrap().unwrap();
+		return super.unwrap();
```

Comment:
Is this change necessary for the upgrade or cleanup?

Reply:
Necessary for the upgrade, otherwise our tests (including `ITCases`) are failing. In upgraded version there is a different structure of nested classes in case of using slices of readonly buffers. Possibly this is a bug fix for a bug, that didn't show up in older netty.

Reply:
But isn't it a bit counter-intuitive that you instantiate a `ReadOnlySlicedNetworkBuffer` with a `NetworkBuffer` and when you call `ReadOnlySlicedNetworkBuffer.unwrap` you don't get a `NetworkBuffer` back but a `SlicedByteBuf`? Thus, you need to know the internals of this class to know that you have to call twice `unwrap` to obtain the `NetworkBuffer`. This shows for example in the `NetworkBufferTest` where we always call `slice.unwrap().unwrap()`.

What exactly did change with the structure of nested classes that this change is necessary?

Reply:
as discussed offline, this is because of change in the implementation of `SlicedByteBuf`.